### ISSUE TITLE
Moderation API

### ISSUE THREAD

**Issue by wjk** [chocolatey/chocolatey.org#223]
I am currently writing a desktop application that integrates with the Chocolatey command-line tool and gallery website. Is there any way to programmatically query the moderation status of a package, through a REST API or similar? (I want my app to show only packages that have successfully passed moderation to the user.)

I would prefer to not have to download the package’s listing page and parse the HTML looking for a moderation tag. If it turns out that I have to, I would probably prepare a PR to this repo that adds an CSS class [here](https://github.com/chocolatey/chocolatey.org/blob/master/chocolatey/Website/Views/Packages/DisplayPackage.cshtml#L115) that has no effect on display, but can be queried for by clients using jQuery or similar to determine the moderation status.

**Comment by gep13** [comment]
@wjk this is not in anyway a direct answer to your question, but I am curious about something...

Is the work that you are doing similar to the work that is already been done in ChocolateyGUI, https://github.com/chocolatey/ChocolateyGUI, if so, we would welcome any contributions that you might want to make there.  Just a thought.

**Comment by wjk** [comment]
@gep13: Yes, I was basically recreating ChocolateyGUI. I was doing it because I disagree with parts of ChocolateyGUI’s visual design and implementation. For example, my implementation has a UI design patterned after that of the Mac App Store (which I feel is very well-designed visually), and is written in Windows Forms instead of WPF. However, if you are open to my suggestions in the ChocolateyGUI repo, I am happy to submit them there instead.

**Comment by wjk** [comment]
Also: To be quite honest, the reason why I didn’t contribute to ChocolateyGUI from the start is because I don’t _quite_ like that the repo’s [CONTRIBUTING](https://github.com/chocolatey/ChocolateyGUI/blob/develop/CONTRIBUTING.md) document states that I should seek approval on the mailing list before I start work on my changes. I would far rather just dive right in, submit a PR early to solicit feedback, and I am perfectly alright with my PRs being criticized or closed because they don’t match the goals for the project. (For an example of the kind of contributing workflow I prefer, look at [my contributions to `CocoaPods/CocoaPods-app`](https://github.com/CocoaPods/CocoaPods-app/pulls?utf8=✓&q=is%3Apr+author%3Awjk).)

**Comment by ferventcoder** [comment]
@wjk The reason we structure it that way is to keep from the alternative happening, where folks are upset b/c we didn't accept their changes. If you are perfectly happy with the criticism, then we are welcome to allowing you to submit that way. We only ask that you file a feature / bug / issue first to form your commits from.

**Comment by wjk** [comment]
Thank you very much, @ferventcoder. I will do just that.

**Comment by ferventcoder** [comment]
I know @gep13 set something up in https://github.com/chocolatey/package-verifier, although I'm not sure it is the best example of how to interact with the interface.

**Comment by ferventcoder** [comment]
The API is an OData feed that hangs off of https://chocolatey.org/api/v2/

**Comment by gep13** [comment]
> design patterned after that of the Mac App Store (which I feel is very well-designed visually)

Do you have any screenshots that you can share?

>written in Windows Forms instead of WPF.

We recently changed ChocolateyGUI away from Windows Forms, in favour of WPF, so it is unlikely that we will be changing back.

>However, if you are open to my suggestions in the ChocolateyGUI repo, I am happy to submit them there instead.

Yes, open to any and all suggestions.

**Comment by gep13** [comment]
>I know @gep13 set something up in https://github.com/chocolatey/package-verifier, although I'm not sure it is the best example of how to interact with the API.

The feed that @ferventcoder mentions was a feed that is going to be used to drive the package-verifier project.  It contains all package versions that are currently in moderation.  In theory, you could query this feed to get the information that you are looking for.

**Comment by wjk** [comment]
@gep13: Here are some screenshots of the Mac App Store I took from my Mac today. Sorry if these are a bit too small for your taste — at full resolution, by my count they would be 2,000 pixels wide!

**Home page**
![Mac App Store Home Page](https://cloud.githubusercontent.com/assets/1915951/10081146/7873168e-62c1-11e5-8944-314002c48156.png)

**Categories List**
![Categories List](https://cloud.githubusercontent.com/assets/1915951/10081158/844253da-62c1-11e5-80ad-54f8be4a3019.png)

**List of Apps in Category**
![Category Content](https://cloud.githubusercontent.com/assets/1915951/10081172/939f7b82-62c1-11e5-9932-37739d9a271c.png)

**App Description Page**
![App Description](https://cloud.githubusercontent.com/assets/1915951/10081191/aaec1598-62c1-11e5-9db7-52b87e9e91ae.png)

**Comment by gep13** [comment]
@wjk thanks for the screenshots.

**Comment by RichiCoder1** [comment]
@wjk I'm planning on revisting the UI, and I'd love to hear some more detail on your thoughts. The current UI for ChocolateyGUI is not great, I absolutely agree, but we lack some metadata data that makes things like Categories (which I love), work and work well. I'd really like to find an inbetween though, and improve the packages list, group, and organization, as well as the general package display and color scheme. Let me know if there's any way I can convince you to assist with ChocolateyGUI.
